### PR TITLE
fix to latest solc

### PR DESCRIPTION
**Pull request by 3esmit**

### CODE REVIEW

File: data.sol [hunk 1]
```diff
@@ -1,4 +1,4 @@
-pragma solidity ^0.5.0;
```
Comment (chriseth) [inline data.sol:1]: Is this a semantic change?

File: patricia.sol [hunk 2]
```diff
@@ -13,40 +13,42 @@ contract PatriciaTree {
...
         D.Label memory k = D.Label(keccak256(key), 256);
         D.Edge memory e = rootEdge;
         bytes32[256] memory siblings;
         uint length;
         uint numSiblings;
         while (true) {
-            var (prefix, suffix) = Utils.splitCommonPrefix(k, e.label);
+            D.Label memory prefix;
+            D.Label memory suffix;
+            (prefix, suffix) = Utils.splitCommonPrefix(k, e.label);
```
Comment (chriseth) [inline patricia.sol]: You should be able to use
```suggestion
            (D.Label memory prefix, D.Label memory suffix) = Utils.splitCommonPrefix(k, e.label);
```

File: patricia.sol [hunk 3]
```diff
@@ -55,7 +57,9 @@ contract PatriciaTree {
             length += prefix.length;
             branchMask |= uint(1) << (255 - length);
             length += 1;
-            var (head, tail) = Utils.chopFirstBit(suffix);
+            uint256 head;
+            D.Label memory tail;
+            (head, tail) = Utils.chopFirstBit(suffix);
```
Comment (chriseth) [inline patricia.sol]: You should be able to use
```suggestion
            (uint head, D.Label memory tail) = Utils.chopFirstBit(suffix);
```

**Review by chriseth — CHANGES_REQUESTED** [review]
Thanks a lot!

I think this is generally fine, but I would recommend some style changes:
 - keep pragma as it was (unless I'm mistaken about something)
 - declare variables with tuple assignment (more places than what I highlighted)

File: data.sol (hunk 1 above)
Comment (3esmit) [reply data.sol:1]: I started using this new pragma because I saw it on the docs. It does not make differece right now, but seems more complete.

Comment (3esmit) [reply data.sol:1]: This is how solidity style guide suggests https://solidity.readthedocs.io/en/v0.5.2/style-guide.html

**Comment by 3esmit** [comment]
I did the changes you requested, and also some things I missed (like hash function as view instead of pure). 
The pragma I changed for how it is defined on Solidity Style Guide. 
Also fixed the ordering of functions (external - public - internal - private).

**Review by chriseth — APPROVED** [review]

**Comment by chriseth** [comment]
Thanks a lot!

**Pull request merged**